regulatory_compliance/monitor.sp

```diff
@@ -44,3 +44,860 @@ control "audit_diagnostic_setting" {
+    group by
+      sub.subscription_id,
+      sub.display_name;
+  EOQ
+}
+
+query "monitor_log_profile_enabled_for_all_regions" {
+  sql = <<-EOQ
+    select
+      p.id as resource,
+      case
+        when p.log_event_location @> '["global", "australiacentral", "australiacentral2", "australiaeast", "australiasoutheast", "brazilsouth", "brazilsoutheast", "canadacentral", "canadaeast", "centralindia", "centralus", "eastasia", "eastus", "eastus2", "francecentral", "francesouth","germanynorth", "germanywestcentral", "japaneast", "japanwest", "jioindiawest", "koreacentral", "koreasouth", "northcentralus", "northeurope",
+        "norwayeast", "norwaywest", "southafricanorth", "southafricawest", "southcentralus", "southeastasia", "southindia", "switzerlandnorth", "switzerlandwest", "uaecentral", "uaenorth", "uksouth", "ukwest", "westcentralus", "westeurope", "westindia", "westus", "westus2", "westus3"]' then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when p.log_event_location @> '["global", "australiacentral", "australiacentral2", "australiaeast", "australiasoutheast", "brazilsouth", "brazilsoutheast", "canadacentral", "canadaeast", "centralindia", "centralus", "eastasia", "eastus", "eastus2", "francecentral", "francesouth","germanynorth", "germanywestcentral", "japaneast", "japanwest", "jioindiawest", "koreacentral", "koreasouth", "northcentralus", "northeurope",
+        "norwayeast", "norwaywest", "southafricanorth", "southafricawest", "southcentralus", "southeastasia", "southindia", "switzerlandnorth", "switzerlandwest", "uaecentral", "uaenorth", "uksouth", "ukwest", "westcentralus", "westeurope", "westindia", "westus", "westus2", "westus3"]' then p.name || ' collect activity logs from all regions.'
+        else p.name || ' not collect activity logs from all regions.'
+      end as reason
+      ${local.tag_dimensions_sql}
+      ${replace(local.common_dimensions_qualifier_sql, "__QUALIFIER__", "p.")}
+      ${replace(local.common_dimensions_qualifier_subscription_sql, "__QUALIFIER__", "sub.")}
+    from
+      azure_log_profile as p
+      left join azure_subscription sub on sub.subscription_id = p.subscription_id;
+  EOQ
+}
+
+query "monitor_application_insights_configured" {
+  sql = <<-EOQ
+    with application_insights as (
+      select
+        subscription_id,
+        count(*) as no_application_insight
+      from
+        azure_application_insight
+      group by
+        subscription_id
+    )
+    select
+      sub.id as resource,
+      case
+        when i.subscription_id is null then 'alarm'
+        else 'ok'
+      end as status,
+      case
+        when i.subscription_id is null then sub.display_name || ' does not have application insights configured.'
+        else sub.display_name || ' has ' || no_application_insight || ' application insights configured.'
+      end as reason
+      ${replace(local.common_dimensions_qualifier_subscription_sql, "__QUALIFIER__", "sub.")}
```

Comment:
@Priyanka-Chatterjee-2000 in this case I don't think we will get `connection_name`.

Reply:
Please check for other places as well.